esp/bindings/http/platform/httpservice.cpp

```diff
@@ -105,6 +105,7 @@ bool CEspHttpServer::rootAuth(IEspContext* ctx)
     EspHttpBinding* thebinding=getBinding();
     if (thebinding)
     {
+        Linked<IInterface> theBindingHolder = dynamic_cast<IInterface*>(thebinding);
```

Comment:
this looks a bit odd/suspicious.
thebinding is of type EspHttpBinding, which does appear to be a IInterface

Reply:
There should also be a check that theBindingHolder != null.  If it should never occur then assertex(theBindingHolder)

Reply:
I found that the rootAuth method is no longer being used, I'll just remove it.